tricycle/_tree_var.py

```diff
@@ -79,26 +78,31 @@ class TreeVar(Generic[T]):
     way. They also provide the additional methods :meth:`being` and
     :meth:`get_in`, documented below.
 
-    Accessing or changing the value of a `TreeVar` outside of a Trio
-    task will raise `RuntimeError`. (Exception: :meth:`get_in` still
-    works outside of a task, as long as you have a reference to the
-    task or nursery of interest.)
-
     .. note:: `TreeVar` values are not directly stored in the
        `contextvars.Context`, so you can't use `Context.get()
        <contextvars.Context.get>` to access them. If you need the value
        in a context other than your own, use :meth:`get_in`.
 
     """
 
-    __slots__ = ("_cvar", "_default")
+    __slots__ = ("_cvar", "_default", "_sync_context_task")
 
     _cvar: contextvars.ContextVar[_TreeVarState[T]]
     _default: T
+    _sync_context_task: trio.lowlevel.Task
 
     def __init__(self, name: str, *, default: T = MISSING):
         self._cvar = contextvars.ContextVar(name)
         self._default = default
+        # Dummy trio task that that simulates the 'current task'
+        # whenever we're running in a synchronous context
+        self._sync_context_task = trio.lowlevel.Task._create(
```

Comment:
Makes sense, I created a separate dummy task class instead so we don't create trio tasks that trio doesn't know about.   
The previous implementation is not correct because of the assumptions I made. I've addressed these 2 as well. 